### ISSUE TITLE
Threshold using multiple features

### ISSUE THREAD

**Issue by ShrimpCryptid** [allen-cell-animated/nucmorph-colorizer#91]
## Use Case

Consider only cells that meet certain criteria defined by one or more feature ranges.

## Acceptance Criteria

- TBD, needs UX design

## Details

- Prerequisite: https://github.com/allen-cell-animated/nucmorph-colorizer/issues/90
- Needs UX
- Dropdown panel? Query box?
- Does the filtering dropdown include every feature type in every dataset? How do we keep ranges across datasets?

**Comment by ShrimpCryptid** [comment]
Discussion with Antoine:
- Fast data visualization, slicing data
ex: What is the effect of environmental conditions on cell speed? -> looking at different criteria (cells w/ 3+ neighbors vs. cells w/ <3 neighbors)

- Looking at cells over time based on speed of cells, for JUST cells 
"Assigning classes to cells using multiple criteria" -> filtering cells by feature value to validate perceived grouping/classes
- Value threshold
  - Dynamic filter (whatever objects CURRENTLY match the filter)
  - Lock a certain subset of tracks at a defined frame and ONLY see those across the rest of the dataset

- Lots of future nice-to-haves, but don't want to slow down development. Ideally something that works NOW, and can be improved later.
- No immediate need to do more than single min + max for a feature
- Ideally no limit to number of filters/thresholds across features :)

**Comment by toloudis** [comment]
what if user changes color-by feature to a feature that is already in our filter set?   how do we handle that?

**Comment by ShrimpCryptid** [comment]
First implementation ideas:
1. pass in an array of sampler2D's into the shader, along with an accompanying min/max range array. 
2. Pre-calculate out-of-range values for all object ID's according to the current filter, and pass it in as a sampler2D.
  1. Naive approach is to use the CPU for this, do operations on a buffer
  2. GPU-based approach could use a buffer texture that is updated with each feature + its min/max every time the filters change.
